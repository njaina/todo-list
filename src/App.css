body{
  background-color: rgb(41, 47, 104);
  padding:5% 30%;
}
.main{
  display: flex;
  color: aliceblue;
}

.response{
  display: flex;
  border: solid 1px;
  margin-top: 5px;
  padding-bottom: 10px;
  border-radius: 6px;
  text-align: center;
  background-color: rgb(26, 30, 68,0.999);
  border-color: rgb(26, 30, 68,0.999);
}

.element{
  width: 15vw;
  margin-top: 10px;
}
.done-elt{
  width: 21vw;
  margin-top: 10px;
}
.check{
  padding-top: 10px;
}
.title{
  text-align: center;
  color: aliceblue;
  margin-bottom: 4%;
}
.todo{
  display: flex;
  flex-direction: column;
  border-color: aliceblue;
}
.input{
  background-color: rgb(37, 18, 61,0.999);
  text-align: center;
  outline: none;
  border: 0;
  height: 20px;
  color: aliceblue;
  width: 17vw;
  height: 4vh;

}
.done{
  margin-left: 20%;
  border-color: aliceblue;
}
.title{
  width: 17vw;
  height: 4vh;
  border: solid 1px;
  border-radius: 9px;
  padding-top: 3%;  
}