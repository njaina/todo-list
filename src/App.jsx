/**class Implemenet extends React.Component(){
    constructor(props){
        super(props);
        this.props={value: 0}
    }

    render(){
        return(
            <>
            <button onClick={()=>this.setState({value:this.state.value+1})} >+</button>
            <div>{this.state.value}</div>
            </>
        )
    }
}*/