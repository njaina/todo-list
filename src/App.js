import { useState } from "react";
import "./App.css"

function App() {
  const [value,setValue]=useState("");
  const [InputValues, setInputValues] = useState([]);
  const [done, setDone] = useState([]);
  
  

  const handleInputChange = (event) => {
    setValue(event.target.value);
  };

  const handleKeyPress = (event) => {
    if (event.key === "Enter") {
      setInputValues([...InputValues,value]);
      setValue('');
    }
  };
  
  const handleCheck = (index) => {
    const doneItem = InputValues[index];
    const newTodos = InputValues.filter((_, i) => i !== index);
    setDone([...done, doneItem]);
    setInputValues(newTodos);
  };
  
  return (
    <div className="main"> 
    <div className="container">
    <div className="todo">
      <div className="title">Todo</div>
      <div>
        <input
          className="input"
          type="text"
          placeholder="New to do"
          value={value}
          onChange={handleInputChange}
          onKeyPress={handleKeyPress}
        />
        <div>
      
        {InputValues.map((todo, index) => (
          <div className="response" >
            <div className="element" key={todo} >{todo}</div>
            <div className="check">
            <input  type="checkbox" onClick={() => handleCheck(index)} /></div>
            </div>
            ))}
        </div>
      </div>
    </div>
    </div>
     
    <div className="container">
    <div className="done">
    <div className="title" >Done</div>
    <div>
      {done.map((todo) => (
        
        <div className="response">
          <div className="done-elt"  key={todo}>{todo}</div>
        </div>
      ))}
    </div>
    </div>
    </div>
    </div>
  );
}

export default App;