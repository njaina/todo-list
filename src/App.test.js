import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

describe('App component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
