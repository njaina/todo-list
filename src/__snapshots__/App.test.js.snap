// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="main"
  >
    <div
      class="container"
    >
      <div
        class="todo"
      >
        <div
          class="title"
        >
          Todo
        </div>
        <div>
          <input
            class="input"
            placeholder="New to do"
            type="text"
            value=""
          />
          <div />
        </div>
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="done"
      >
        <div
          class="title"
        >
          Done
        </div>
        <div />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
